backend/src/boundary_stats.rs

```diff
@@ -9,12 +8,52 @@ pub struct BoundaryStats {
 }
 
 impl BoundaryStats {
-    pub fn new(polygon: &Polygon) -> Self {
-        // Convert from m^2 to km^2. Use unsigned area to ignore polygon orientation.
-        let area_km2 = polygon.unsigned_area() / 1_000_000.0;
-        // TODO: SIMD
-        let mut rng = WyRand::new_seed((area_km2 * 1000000.0) as u64);
-        let simd = f64::random(&mut rng) * 100.0;
-        Self { area_km2, simd }
+    pub fn new(polygon: &Polygon, population_zones: Option<&[PopulationZone]>) -> Self {
+        // Use unsigned area to ignore polygon orientation.
+        let area_meters = polygon.unsigned_area();
+
+        // TODO: review this methodology
+        // area proportional accumulation:
+        // If  3/4 the boundary is in a zone with simd 100:   0.75 * 100 = 75
+        // and 1/4 the boundary is in a zone with  simd 40:  +0.25 *  40 = 10
+        //                  than the computed simd will be:          sum = 85
+        // TODO: optimize this - populationZone should at least be prepared geometries, and
+        // TODO: maybe also introduce an overall RTree to get intersection candidates quickly
+        let mut simd = 0.0;
+        if let Some(population_zones) = population_zones {
+            for population_zone in population_zones {
+                if population_zone.geometry.intersects(polygon) {
```

Comment:
I don't know much about the implementations of these -- is there lots of "wasted work" to first check for intersections, then calculate exactly what it is? Presumably the first is much faster, and something like a bounding box overlap check isn't a useful alternate first pass?